subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt

```diff
@@ -361,9 +378,9 @@ class ResidualProgramCompiler(
     }
 
     private
-    fun MethodVisitor.emitPluginRequestCollectorCreateSpecFor(plugins: Program.Plugins) {
+    fun MethodVisitor.emitPluginRequestCollectorCreateSpecFor(plugins: Program.Plugins?) {
         ALOAD(Vars.PluginRequestCollector)
-        LDC(plugins.fragment.lineNumber)
+        LDC(plugins?.fragment?.lineNumber ?: 0)
```

Comment:
@bamboo This was a kludge that I wasn't proud of. Is there a better way to do this than what I did here?

Reply:
We could use a separate branch and template when the `plugins` block is not there, one that wouldn't require a collector.